### PR TITLE
Add new post link to Query

### LINKED ISSUE #23461: Start a new post from Query Loop
The query loop block should have an easy way to write new posts. I've noticed some users try to write posts directly in the content of the editor, either above or beneath what "latest posts" or the "query" shows. This seems like a great place to tie the loops, teach users, and guide them around creating content in the relevant places.

It's similar to how the navigation block should have a way to add a new page. Of course, once Query supports different custom post types, it should be routed accordingly.

**Examples:**

- We could add a "write new post" in the toolbar of the Query Loop block.
- In the sidebar within the description area of the block, including a description "The loop displays a list of your posts. (Link)Write New Post."

image

### PR DESCRIPTION
**Pull request by ntsekouras**
## Description
Resolves: https://github.com/WordPress/gutenberg/issues/23461

## How has this been tested?
Manually

## Types of changes
Non breaking changes.

### CODE REVIEW

File: packages/block-library/src/query/edit/query-inspector-controls.js [hunk 1]
```diff
@@ -143,8 +144,21 @@ export default function QueryInspectorControls( {
...
 		return onChangeDebounced.cancel;
 	}, [ querySearch, onChangeDebounced ] );
+
+	const CreateNewPostLink = ( { type } ) => {
+		const newPostUrl = addQueryArgs( 'post-new.php', {
+			post_type: type,
+		} );
+		return (
+			<div className="wp-block-query__create-new-link">
+				<a href={ newPostUrl }> { __( 'Create a new post' ) }</a>{ ' ' }
```
Comment (ntsekouras) [inline packages/block-library/src/query/edit/query-inspector-controls.js:154]: Needs better handling of translations here. Should I wrap all the text in a link like other cases in GB (ex `Learn more about manual excerpts`).

File: packages/block-library/src/query/edit/query-inspector-controls.js [hunk 2]
```diff
@@ -143,8 +150,24 @@ export default function QueryInspectorControls( {
...
+					__( '<a>Create a new post</a> for this feed.' ),
+					// eslint-disable-next-line jsx-a11y/anchor-has-content
+					{ a: <a href={ newPostUrl } /> }
+				) }
+			</div>
+		);
+	};
 	return (
 		<InspectorControls>
+			<CreateNewPostLink type={ postType } />
```
Comment (mcsf) [inline packages/block-library/src/query/edit/query-inspector-controls.js:170]: I would either:

* inline the whole element tree, i.e. `<InspectorControls><div className="...create-new-link">...`
* or move the definition of `CreateNewPostLink` one level up, to the module

**Review by mcsf — APPROVED** [review]
Looks good insofar as it implements https://github.com/WordPress/gutenberg/issues/23461#issuecomment-736144629. Of course, we can improve the interaction as we move along.

File: packages/block-library/src/query/edit/query-inspector-controls.js (hunk 2 above)
Comment (ntsekouras) [reply packages/block-library/src/query/edit/query-inspector-controls.js:171]: I moved it one level up.

**Pull request merged**